### PR TITLE
Add some Rebar3 plugin debug logging

### PR DESCRIPTION
**Pull request by erszcz**
This should help in cases like @ilya-klyuchnikov's #530, where it's not clear what options the Rebar3 plugin runs Gradualizer with and on what files. We have to use `DEBUG=1 rebar3 gradualizer` to see the debug printouts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ To run Gradualizer from rebar3, add it as a plugin in your `rebar.config`:
 
 ```erlang
 {plugins, [
-  {gradualizer, {git, "git@github.com:josefs/Gradualizer.git", {branch, "master"}}}
+  {gradualizer, {git, "https://github.com/josefs/Gradualizer.git", {branch, "master"}}}
```
Comment (zuiderkwast) [inline README.md:94]: Why these changes?

File: src/rebar_prv_gradualizer.erl [hunk 2]
```diff
@@ -17,6 +17,9 @@
 -define(PROVIDER, gradualizer).
 -define(DEPS, [compile]).
 
+%% This is borrowed from the private upstream rebar.hrl
+-define(DEBUG(Str, Args), rebar_log:log(debug, Str, Args)).
```
Comment (zuiderkwast) [inline src/rebar_prv_gradualizer.erl:21]: Does this give debugging printouts with some command line option, like `rebar3 --debug gradualizer` or something?

File: README.md (hunk 1 above)
Comment (erszcz) [reply README.md:94]: `git://` scheme was dropped completely by GitHub, so I wanted to update the docs. On the occasion it doesn't hurt to unify all the examples to use the same scheme.

File: src/rebar_prv_gradualizer.erl (hunk 2 above)
Comment (erszcz) [reply src/rebar_prv_gradualizer.erl:21]: Yes, that's what the `DEBUG=1 ...` from the PR description does.

File: README.md (hunk 1 above)
Comment (zuiderkwast) [reply README.md:94]: `git@github.com` was not dropped. Only git:// (without username) was dropped IUUC.

File: src/rebar_prv_gradualizer.erl (hunk 2 above)
Comment (zuiderkwast) [reply src/rebar_prv_gradualizer.erl:21]: Ahh 🤦‍♂️PR description, too obvious
